test-case/check-alsabat.sh

```diff
@@ -71,16 +73,16 @@ dlogi "check the PCMs before alsabat test"
 # different PCMs may support different audio formats(like samplerate, channel-counting, etc.).
 # use plughw to do the audio format conversions. So we don't need to specify them for each PCM.
 dlogc "alsabat -Pplug$pcm_p --standalone -n $frames -F $frequency"
-alsabat -Pplug$pcm_p --standalone -n $frames -F $frequency &
+timeout --kill-after 3 20 alsabat \
```

Comment:
a timeout watchdog for the alsabat? I think we have clean up for alsabat aplay arecord thread in exit.

Reply:
> a timeout watchdog for the alsabat? I think we have clean up for alsabat aplay arecord thread in exit.

No test should leave random processes behind, each test should clean up after itself because only the test knows exactly what was started. Cleaning up in the exit function is great to give the next test a chance but finding processes still hung at the exit stage should be treated as a test failure.

The `timeout` and  `wait` are unrelated to each other.

`wait` is the only way to collect errors from background commands. Every background command should have a corresponding wait.

The `timeout` is just me being  paranoid and be prepared in case alsabat gets stuck. If there's no precedence of alsabat ever getting stuck then it's probably overkill and I can remove it.

Reply:
> The timeout and wait are unrelated to each other.

Actually they are related, sorry for the confusion: `alsabat -P &` getting stuck  would stall the test only when there is a `wait`. But `alsabat -C` is not in the background so it could stall the test too. To be consistent there should be a `timeout` on either both `-P` and `-C` or none.

I removed the `timeout`, there are already 3 different commits in this PR that's enough for one PR.

Reply:
@marc-hb Got that, you also want to get the logs from alsabat.